src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -53,14 +66,12 @@ public static class ConcurrencyUtilities
                 {
                 }
 
-                using (fs)
-                {
-                    return await action(token);
-                }
+                return fs;
```

Comment:
It's possible that fs will miss the dispose here if token.ThrowIfCancellationRequested(); above throws. I tried cleaning this method up when adding delete on close and perf was dramatically worse.